### PR TITLE
Fix UITransactionDlg rendering bug

### PR DESCRIPTION
**Pull request by UTengine**
This fixes the NPC icons being scattered around the 12 pages. 
And not being able to purchase some of them in the selling group list.

### Description
This fixes the following issue/ticket:
https://github.com/ko4life-net/ko/issues/203

💔Thank you!

### CODE REVIEW

**Comment by UTengine** [comment]
PS/NOTE: Although this fixes npc selling icons I want this to be double tested and checked thoroughly, I don't feel okay with the loops especially if end user ends up selling an item slot that he didn't intend to sell which is a big disaster lol.

**Review by stevewgr — CHANGES_REQUESTED** [review]
Hi @UTengine, thanks for the PR.
The changes in the PR are wrong. We should not revert back to the old outer-loop that was fixed in here: https://github.com/ko4life-net/ko/pull/33

The only mistake here is that the `k++` have to take place in the end of the loop. I was able to reproduce on my end.

Could you please fix your PR?

File: src/game/UITransactionDlg.cpp [hunk 1]
```diff
@@ -302,6 +304,7 @@ void CUITransactionDlg::EnterTransactionState() {
         spItem->iDurability = pItem->siMaxDurability + pItemExt->siMaxDurability;
 
         m_pMyTrade[j][k] = spItem;
+		k++;
```
[the remark this answers is not in the document]
Comment (stevewgr) [reply src/game/UITransactionDlg.cpp]: As discussed in Discord channel, Tahsin fixed the requested changes. Resolving.

**Review by stevewgr — APPROVED** [review]

**Comment by stevewgr** [comment]
Merged in here: https://github.com/ko4life-net/ko/pull/212

because the last push from @UTengine broke the PR, so I had to cherry-pick and separate commits.

**Pull request closed without merging**